05.ls/ls.rb

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'date'
+require 'etc'
+require 'optparse'
+
+STRING_TO_CHAR_FILE_TYPE = {
+  'file' => '-',
+  'directory' => 'd',
+  'characterSpecial' => 'c',
+  'blockSpecial' => 'b',
+  'fifo' => 'p',
+  'link' => 'l',
+  'socket' => 's',
+  'unknown' => 'u'
+}.freeze
+
+NUMERIC_TO_SYMBOLIC_PERMISSION = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
+
+def main
+  options = ARGV.getopts('alr')
+
+  file_names = options['a'] ? Dir.glob('*', File::FNM_DOTMATCH) : Dir.glob('*')
+  file_names = file_names.sort
+  file_names = file_names.reverse if options['r']
+
+  puts options['l'] ? files_in_long_format(file_names) : files(file_names)
```

Comment:
`files` メソッドのほうに対称性を持たせておくと何をするメソッドなのか分かりやすいかも。

```suggestion
  puts options['l'] ? files_in_long_format(file_names) : files_in_short_format(file_names)
```